### ISSUE TITLE
could not find any matching episode

### ISSUE THREAD

**Issue by ltsj** [crunchy-labs/crunchy-cli#89]
**Describe the bug**
This error was encountered today and has never happened before
When I download it says:
=> Failed to parse url 1
=> An error occurred: failed to get episodes: could not find any matching episode

**To Reproduce**
Steps / command to reproduce the behavior:
```
$ crunchy.exe --lang en-US archive -l ja-JP https://www.crunchyroll.com/series/GVDHX8QNW/chainsaw-man
```

**Screenshots**
![image](https://user-images.githubusercontent.com/30733960/209295022-695034e3-3d64-435d-bb38-6081a24f1c5a.png)

**Client (please complete the following information):**
 - OS: Windows
 - Version v2.3.6

**Comment by adracea** [comment]
Hey there, this seems to be a recent change in crunchyroll? I've looked at quite a few different series and seemingly the ja-JP locale has disappeared or is wrongly reported as en-US which creates quite some confusion. If you are sure that the URL you are using is proper for the ja-JP series, you can skip specifying the locale and it should work fine.

Otherwise @ByteDream  for awareness.

**Comment by ltsj** [comment]
> 

thanks i will try :)

**Comment by ltsj** [comment]
😥I'm not sure if this is another crunchyroll change and when I try to log in now it says:
`[err] An error occurred: error for endpoint https://beta-api.crunchyroll.com/accounts/v1/me (406): invalid json response: invalid character 'N' after top-level value`

**Comment by adracea** [comment]
Can you try one of the dev releases ? Anything above 3.0.0 should work fine. The cli has changed from using go to using rust and the 2.x releases are quite outdated at this point.

**Comment by ltsj** [comment]
I tried 3.x and it keeps showing the progress bar at 100%, but there is no follow-up action,
and I've been waiting for a long time and still nothing...
```
:: Downloading Recap Episode: Lucifer and the Biscuit Hammer, First Half to 'Recap Episode Lucifer and the Biscuit Hammer, First Half.mkv'
::      Episode: S01E00
::      Audio: en-US (primary), 
::      Subtitle: ar-SA, de-DE, en-US, es-419, es-ES, fr-FR, it-IT, pt-BR, ru-RU
::      Resolution: 1920x1080
::      FPS: 23.97
:: Download en-US 1456.2 MiB  10.26 MiB/s [##########################################################################################################################################################################################] 100%
```
When I try to use `-v` it shows:
```
error: Found argument '-v' which wasn't expected, or isn't valid in this context

  If you tried to supply '-v' as a value rather than a flag, use '-- -v'

```

**Comment by adracea** [comment]
> I tried 3.x and it keeps showing the progress bar at 100%, but there is no follow-up action, and I've been waiting for a long time and still nothing...
> 
> ```
> :: Downloading Recap Episode: Lucifer and the Biscuit Hammer, First Half to 'Recap Episode Lucifer and the Biscuit Hammer, First Half.mkv'
> ::      Episode: S01E00
> ::      Audio: en-US (primary), 
> ::      Subtitle: ar-SA, de-DE, en-US, es-419, es-ES, fr-FR, it-IT, pt-BR, ru-RU
> ::      Resolution: 1920x1080
> ::      FPS: 23.97
> :: Download en-US 1456.2 MiB  10.26 MiB/s [##########################################################################################################################################################################################] 100%
> ```
> 
> When I try to use `-v` it shows:
> 
> ```
> error: Found argument '-v' which wasn't expected, or isn't valid in this context
> 
>   If you tried to supply '-v' as a value rather than a flag, use '-- -v'
> ```

Hey so, I can replicate the behavior of the stuck download, it seems to be a sort of hit/miss situation sometimes with *something* getting stuck.

As for the `-v` option : you have to specify it in front of everything else like so:
`crunchy-cli.exe -v download $(EP_URL) -r best -l en-US -s en-US`
`crunchy-cli.exe -v archive $(SERIES/SEASON_URL) -r best -l en-US -s en-US`

**Comment by ltsj** [comment]
> 

Thanks! I will keep trying :)

**Comment by adracea** [comment]
@ltsj are you still able to replicate the stuck downloads ? I just tried this again in the debugger and for some reason it's worked flawlessly for the past 20 minutes. I'm wondering if maybe it was just a small network hiccup on the crunchyroll side ?_?

**Comment by ltsj** [comment]
@adracea So far, I can't use the 3.x version to download,it will always be stuck at 100% progress

**Comment by adracea** [comment]
Seems like it might be hitting on some bandwidth based rate limiting from crunchyroll, I am now able to replicate when using `-r best` quality . It might be related to some fixes we applied between dev2 and dev3 releases. Could you give a shot to :
https://github.com/crunchy-labs/crunchy-cli/releases/tag/v3.0.0-dev.2 ?

**Comment by ltsj** [comment]
> Seems like it might be hitting on some bandwidth based rate limiting from crunchyroll, I am now able to replicate when using quality . It might be related to some fixes we applied between dev2 and dev3 releases. Could you give a shot to : https://github.com/crunchy-labs/crunchy-cli/releases/tag/v3.0.0-dev.2 ?`-r best`

[v3.0.0-dev.2](https://github.com/crunchy-labs/crunchy-cli/releases/tag/v3.0.0-dev.2) can be downloaded normally

**Comment by adracea** [comment]
@ltsj  I would also recommend upgrading the ffmpeg binary to one of the newer releases. I've had various issues stemming from it especially on `Windows`.

**Comment by IchBinLeoon** [comment]
https://github.com/crunchy-labs/crunchyroll-rs/issues/3

**Comment by ByteDream** [comment]
This should be fixed in [v3.0.0-dev.5](https://github.com/crunchy-labs/crunchy-cli/releases/tag/v3.0.0-dev.5). This adds a workaround for Crunchyrolls broke and crappy api results 🙃.
